railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -13,7 +13,9 @@ def initialize(app, config, paths)
 
       def build_stack
         ActionDispatch::MiddlewareStack.new do |middleware|
-          middleware.use ::ActionDispatch::HostAuthorization, config.hosts, **config.host_authorization
+          unless config.hosts.empty?
+            middleware.use ::ActionDispatch::HostAuthorization, config.hosts, **config.host_authorization
+          end
```

Comment:
Do you think it would be worth including a test for the conditional, similar to [this test](https://github.com/rails/rails/blob/d05395ce8ab8b769dbab7fd547580ed7ca94c051/railties/test/application/middleware_test.rb#L216-L220)?

Reply:
Done!